src/libexpr/primops.cc

```diff
@@ -1120,15 +1120,18 @@ drvName, Bindings * attrs, Value & v)
         vomit("processing attribute '%1%'", key);
 
         auto handleHashMode = [&](const std::string_view s) {
-            if (s == "recursive") ingestionMethod = FileIngestionMethod::Recursive;
-            else if (s == "flat") ingestionMethod = FileIngestionMethod::Flat;
-            else if (s == "text") {
-                experimentalFeatureSettings.require(Xp::DynamicDerivations);
-                ingestionMethod = TextIngestionMethod {};
-            } else
+            if (s == "recursive") {
+                // back compat, new name is "nar"
+                ingestionMethod = FileIngestionMethod::Recursive;
```

Comment:
Don't we instead want to accept `recursive as a file ingestion method for consistency?

Reply:
I am not sure what you mean, I think change `ContentAddressMethod::parse` so this one is accepted everywhere (e.g. the new CLI too)?

Reply:
Yes, that's what I meant

Reply:
I forget whether we discussed this before, but I do want to over time steer people to the new name (which is clearer) in the new CLI.

Reply:
(Also with git support, which I hope to mere very soon, NAR isn't the only "recursive" format anyways.)